runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/artifact/DfsArtifactRetrievalService.java

```diff
@@ -0,0 +1,135 @@
+      StreamObserver<ArtifactApi.GetManifestResponse> responseObserver) {
+    try {
+      ArtifactApi.ProxyManifest proxyManifest = loadManifestFrom(request.getRetrievalToken());
+      ArtifactApi.GetManifestResponse response =
+          ArtifactApi.GetManifestResponse
+              .newBuilder()
+              .setManifest(proxyManifest.getManifest())
+              .build();
+      responseObserver.onNext(response);
+      responseObserver.onCompleted();
+    } catch (Exception e) {
+      responseObserver.onError(e);
+    }
+  }
+
+  @Override
+  public void getArtifact(
+      ArtifactApi.GetArtifactRequest request,
+      StreamObserver<ArtifactApi.ArtifactChunk> responseObserver) {
+    try {
+      ArtifactApi.ProxyManifest proxyManifest = loadManifestFrom(request.getRetrievalToken());
+      // validate that name is contained in manifest and location list
+      boolean containsArtifactName =
+          ImmutableList.copyOf(proxyManifest.getManifest().getArtifactList())
+              .stream()
+              .anyMatch(metadata -> metadata.getName().equals(request.getName()));
+      if (!containsArtifactName)  {
+        throw new ArtifactNotFoundException(request.getName());
+      }
+      // look for file at URI specified by proxy manifest location
+      ImmutableList<ArtifactApi.ProxyManifest.Location> locationList =
+          ImmutableList.copyOf(proxyManifest.getLocationList());
+      ArtifactApi.ProxyManifest.Location location =
+          locationList
+              .stream()
+              .filter(loc -> loc.getName().equals(request.getName()))
+              .findFirst()
+              .orElseThrow(() -> new ArtifactNotFoundException(request.getName()));
+      ResourceId artifactResourceId =
+          FileSystems.matchNewResource(location.getUri(), false /* is directory */);
+      ReadableByteChannel artifactByteChannel = FileSystems.open(artifactResourceId);
+      ByteBuffer byteBuffer = ByteBuffer.allocate(artifactChunkSizeBytes);
+      while (artifactByteChannel.read(byteBuffer) > -1) {
+        ByteString data = ByteString.copyFrom(byteBuffer);
+        byteBuffer.clear();
+        ArtifactApi.ArtifactChunk artifactChunk =
+            ArtifactApi.ArtifactChunk.newBuilder().setData(data).build();
+        responseObserver.onNext(artifactChunk);
+      }
+      responseObserver.onCompleted();
+    } catch (Exception e) {
+      responseObserver.onError(e);
+    }
+
+  }
+
+  @Override
+  public void close() throws Exception {
+
+  }
+
+  private ArtifactApi.ProxyManifest loadManifestFrom(String retrievalToken) throws Exception {
+    // look for file at $retrieval_token/$manifestName
+    ResourceId manifestResourceId =
+        FileSystems.matchNewResource(retrievalToken, false /* is directory */);
+    InputStream manifestStream = Channels.newInputStream(FileSystems.open(manifestResourceId));
+    ArtifactApi.ProxyManifest proxyManifest =  ArtifactApi.ProxyManifest.parseFrom(manifestStream);
```

Comment:
The staging service stores this an JSON. We should be consistent here. @angoenka